### PR TITLE
Implemented job saving in database and local client to backend communication

### LINKED ISSUE #163: implement saving jobs in DB
Jobs should be put into the database, so that they can be retrieved from the local client from there.

---

### :thought_balloon: &nbsp; Hints & Thoughts

- create model for a junction object connecting a local client id with a user id
- create a model for a job with a userId, SSoT_Id, passed parameters and a status field
- the junction object should be used by the local client to check for the user, to which it is linked and retrieve the jobs from the job collection based on that

---

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] Necessary models should be created
- [ ] Adjust routes for the local client to retrieve pending jobs for the linked user
- [ ] the functionality is tested with a local DB copy

### LINKED ISSUE #166: allow local client to connect to web interface
As a end user I would like to pass the local client a userId and for it to run my jobs when I enqueue them.

---

### :question: &nbsp; Additional Questions to Answer / Consider

---

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] Should allow for inputting a userId into the local client
- [ ] create a new junction object linking the local client and the userId in the database

### PR DESCRIPTION
**Pull request by SanJSp**
Fixes #166 and fixes #163

## Changes
First, this PR implements the routes, models, and controllers for sessions and jobs. Sessions are database objects that connect a local client to a user. Jobs are database objects that connect a single source of truth to a user, spiced up with a status field and parameters.

Additionally, this implementation expands the functionality of our local client. Now a local client has a client id (currently still hardcoded) and a saved session. The user is questioned in the beginning, if he/she like to continue with the last session or start a new one. Starting a new one adds a new session to the database with the client id and the prompted user_id. Using the session-id the local client gets the session from the database and extracts the user id from that.
With the user_id the local client gathers all jobs that are waiting for execution for that user. For each, the local client uses the robot_id /ssot_id to get the executable robot code and executes it immediately.

After each job has been worked on, the local_client polls every 10 seconds for new jobs giving a quick status over the command line each time. This continues until the local client is interrupted.

### Zum Erklären der Datenbank (mit Ton)
https://user-images.githubusercontent.com/38314662/111190966-e60c3080-85b7-11eb-8d4b-6a501e2e8062.mp4

### Demo des lokalen clients (mit Ton)
https://user-images.githubusercontent.com/38314662/111191927-d50fef00-85b8-11eb-8824-b184f909cdcc.mp4

## Checklist before merge

**Developer's responsibilities**
* [ ] **Assign** one or two developers
* [ ] **Change code** if reviewer(s) has/have requested it
* [ ] **Pull request build** has passed
* [ ] **tested locally** (in at least chrome & firefox if frontend)
* [ ] updated the **documentation**
* [ ] added **tests** where necessary

### CODE REVIEW

File: client/src/components/content/RobotContainer/RobotContainer.jsx [hunk 1]
```diff
@@ -2,59 +2,84 @@
 import React, { useState } from 'react';
 import { Col, Row, Typography } from 'antd';
 import { PlayCircleOutlined, EditOutlined } from '@ant-design/icons';
-import PropTypes from 'prop-types'
+import PropTypes from 'prop-types';
 import styles from './RobotContainer.module.css';
-import {changeSSOTName} from '../../../api/SSOTretrieval';
+import { changeSSOTName } from '../../../api/SSOTretrieval';
+import setRobotJob from '../../../api/jobs';
+import isBotExecutable from '../../../utils/botExecution';
```
Comment (kej-jay) [inline client/src/components/content/RobotContainer/RobotContainer.jsx:9]: use robot instead of bot

File: client/src/utils/botExecution.jsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// is a dummy function
+const isBotExecutable = () => {
+    if (Math.random() > 0.49) {
```
Comment (kej-jay) [inline client/src/utils/botExecution.jsx:3]: Pretty sure, this is not a good thing to do... What is its functionality? Why is a random number needed for it that seems to be a 50/50 chance generator

**Review by kej-jay — CHANGES_REQUESTED** [review]
I stopped mid way in the review. There are many files, which are not yet in DEV. Please merge your initial branch first and/or rebase this branch, so that I don't review files which are already in review

**Review by WolfgangDaniel — CHANGES_REQUESTED** [review]
Unfortunately, we probably won't need those changes anymore as we will use socket.io going forward.

**Comment by SanJSp** [comment]
> 
> 
> Unfortunately, we probably won't need those changes anymore as we will use socket.io going forward.

I think we do, as #162 is only a spike limited to 10 hours of working time and no finished implementation is expected. Otherwise, the issue #166 would be marked as dependent on #162 which it isn't.

**Comment by SanJSp** [comment]
PR closed because too big and #166 will be implemented by #162

**Pull request closed without merging**